tsdb/cursor.go

```diff
@@ -1,35 +1,22 @@
 package tsdb
 
 import (
-	"bytes"
 	"container/heap"
-)
-
-// Direction represents a cursor navigation direction.
-type Direction bool
+	"encoding/binary"
+	"sort"
+	"strings"
 
-const (
-	// Forward indicates that a cursor will move forward over its values.
-	Forward Direction = true
-	// Reverse indicates that a cursor will move backwards over its values.
-	Reverse Direction = false
+	"github.com/influxdb/influxdb/influxql"
 )
 
-func (d Direction) String() string {
-	if d.Forward() {
-		return "forward"
-	}
-	return "reverse"
-}
-
-// Forward returns true if direction is forward
-func (d Direction) Forward() bool {
-	return d == Forward
-}
+// EOF represents a "not found" key returned by a Cursor.
+const EOF = int64(-1)
 
-// Forward returns true if direction is reverse
-func (d Direction) Reverse() bool {
-	return d == Reverse
+// Cursor represents an iterator over a series.
+type Cursor interface {
+	Seek(seek int64) (key int64, value interface{})
```

Comment:
I might change it to `SeekTo()` for now. It mostly bothers me that `io.Seeker` is in `go vet` when it's so rarely used.